app/uk/gov/hmrc/merchandiseinbaggage/repositories/crypto/DeclarationRepositoryCryptoImpl.scala

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.merchandiseinbaggage.repositories.crypto
+
+import uk.gov.hmrc.merchandiseinbaggage.model.api._
+import uk.gov.hmrc.merchandiseinbaggage.repositories.{DeclarationRepository, DeclarationRepositoryImpl}
+
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.{ExecutionContext, Future}
+
+@Singleton
+class DeclarationRepositoryCryptoImpl @Inject()(dr: DeclarationRepositoryImpl, cryp: DeclarationCrypto)(implicit ec: ExecutionContext)
```

Comment:
with this new repo class, not sure if we still need the old one `DeclarationRepositoryImpl` ?

Reply:
This is a wrapper around DeclarationRepositoryImpl and only encrypts/decrypts declarations, before passing them on. Have renamed it to include Wrapper to make this more obvious.